src/components/date_picker/super_date_picker/super_update_button.tsx

```diff
@@ -1,29 +1,34 @@
-import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { Component, MouseEventHandler, Ref } from 'react';
 import classNames from 'classnames';
 
 import { EuiButton } from '../../button';
 import { EuiI18n } from '../../i18n';
-import { EuiToolTip } from '../../tool_tip';
-
-export class EuiSuperUpdateButton extends Component {
-  static propTypes = {
-    needsUpdate: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    onClick: PropTypes.func.isRequired,
-    /**
-     * Passes props to `EuiToolTip`
-     */
-    toolTipProps: PropTypes.object,
-  };
+import { EuiToolTip, EuiToolTipProps } from '../../tool_tip';
+
+export interface EuiSuperUpdateButtonProps {
+  className?: string;
+  isDisabled: boolean;
+  isLoading: boolean;
+  needsUpdate: boolean;
+  onClick: MouseEventHandler<HTMLButtonElement>;
+
+  /**
+   * Passes props to `EuiToolTip`
+   */
+  toolTipProps?: EuiToolTipProps;
+}
 
+export class EuiSuperUpdateButton extends Component<EuiSuperUpdateButtonProps> {
   static defaultProps = {
     needsUpdate: false,
     isLoading: false,
     isDisabled: false,
   };
 
+  _isMounted = false;
+  tooltipTimeout: NodeJS.Timer | undefined;
```

Comment:
I hate this one :old-man-yells-at-clouds: . We've dealt with this [elsewhere](https://github.com/elastic/eui/blob/6f9afe3b569b7aa75b8b550b6494487e545b4d89/src/components/delay_hide/delay_hide.tsx#L82) by casting the `setTimeout` delay `as number | undefined` , to retain the window.setTimeout typing

```suggestion
  tooltipTimeout: number | undefined;
```

Reply:
https://gph.is/2cPGi7y

no problem, heh.  I get it.  These things can be a not worth the trouble.

https://github.com/elastic/eui/pull/2891/commits/a6006583c607c5b2f46efd585eb28505bd90e993